lib/events.js

```diff
@@ -20,6 +20,17 @@ EventEmitter.prototype._maxListeners = undefined;
 // added to it. This is a useful default which helps finding memory leaks.
 EventEmitter.defaultMaxListeners = 10;
 
+function ensureEvents(emitter) {
+  if (!emitter._events ||
+      emitter._events === Object.getPrototypeOf(emitter)._events) {
```

Comment:
@bnoordhuis in this example instances of A and B would have own `_events` because of super constructor calls. If you omit those it should still be fine. `Object.getPrototypeOf(emitter)._events` returns inherited `_events` if immediate prototype has none.